src/fsharp/CompileOps.fs

```diff
@@ -5205,11 +5207,11 @@ module private ScriptPreprocessClosure =
         let allRootDiagnostics = allRootDiagnostics |> List.filter (fst >> isRootRange)
         
         let result : LoadClosure = 
-            { SourceFiles = List.groupByFirst sourceFiles
-              References = List.groupByFirst references
+            { SourceFiles = List.groupBy fst sourceFiles |> List.map (map2Of2 (List.map snd))
```

Comment:
What are these doing? What is `map2of2`?

Reply:
@TIHan in ```visualfsharp\src\fsharp\lib.fs```
``` let map2Of2 f (a1,a2)       = (a1,f a2) ```

Reply:
Yeah, I removed the groupByFirst helper (which was a reimplementation of groupBy) but then had to add these since the returned results were slightly different - I should have added a new groupByFirst helper implemented in terms of groupBy